ethcore/src/engines/tendermint/mod.rs

```diff
@@ -491,14 +497,15 @@ impl Engine for Tendermint {
 	fn verify_block_basic(&self, header: &Header, _block: Option<&[u8]>) -> Result<(), Error> {
 		let seal_length = header.seal().len();
 		if seal_length == self.seal_fields() {
+			let proposal_len = header.seal()[1].len();
 			let signatures_len = header.seal()[2].len();
-			if signatures_len >= 1 {
+			if (proposal_len == 1) ^ (signatures_len == 1) {
```

Comment:
I don't really get that change here.
So the block is valid either if:
1. there is one proposal and the number of signatures is not one (so 0 or 2 sigs is ok)
2. or there is one signature and the number of proposals is not one (so 0 or 2 proposals is ok),

Why is that correct? (Shouldn't it be either (a proposal and 0 sigs) or (a signature and 0 proposals)?)